pkg/controllers/plugin/plugin_workload_status_controller.go

```diff
@@ -196,24 +230,43 @@ func getStatusPayload(ctx context.Context, releaseStatus *ReleaseStatus, cl clie
 			log.FromContext(ctx).Error(err, "Error getting replicaset", "name", objName, "pluginName", releaseStatus.ReleaseName)
 			return
 		}
-		releaseStatus.Replicas += remoteObject.Status.Replicas
-		releaseStatus.ReadyReplicas += remoteObject.Status.ReadyReplicas
-		releaseStatus.AvailableReplicas += remoteObject.Status.AvailableReplicas
+		status.Replicas += remoteObject.Status.Replicas
+		status.ReadyReplicas += remoteObject.Status.ReadyReplicas
+		status.AvailableReplicas += remoteObject.Status.AvailableReplicas
 
 		if !isPayloadReadyRunning(remoteObject) {
-			releaseStatus.ReleaseOK = false
-			releaseStatus.Message = "Not all pods are running in the ReplicaSet"
+			status.Ready = false
+			status.Message = fmt.Sprintf("%s/%s", gvk.Kind, objName)
+		}
+	case "Job":
+		remoteObject := &batchv1.Job{}
+		if err := cl.Get(context.TODO(), types.NamespacedName{Name: objName, Namespace: objNamespace}, remoteObject); err != nil {
```

Comment:
The `context.TODO` can be replaced by passing in the context passed into `Reconcile(..)`.